vhost-device-gpu/src/backend/virgl.rs

```diff
@@ -159,16 +163,25 @@ impl VirglRendererAdapter {
             fence_state.clone(),
         ));
 
-        let renderer = VirglRenderer::init(virglrenderer_flags, fence_handler, None)
-            .expect("Failed to initialize virglrenderer");
-        Self {
+        // Use the GPU device FD if provided (already opened and validated at startup)
+        let render_server_fd = config
+            .flags()
+            .render_server_fd
+            .as_ref()
+            .map(|fd| fd.try_clone())
+            .transpose()
+            .map_err(|e| io::Error::other(format!("Failed to clone GPU device FD: {e}")))?;
+
+        let renderer = VirglRenderer::init(virglrenderer_flags, fence_handler, render_server_fd)
```

Comment:
What happens here if there is no GPU available in the system? (I mean when the user does not specify any path)

IIUC this will fail (due to lazy initialization) when the first request from the VM arrives, so I don't understand why we added complexity to open the path as soon as possible, if we then have the same problem in the most common use case.

I'm confused, but it's pre-existing, so it's fine, it's not blocking, it's just very strange.